OpenRA.Mods.Common/Traits/Demolishable.cs

```diff
@@ -68,16 +68,18 @@ void IDemolishable.Demolish(Actor self, Actor saboteur, int delay, BitSet<Damage
 
 		void ITick.Tick(Actor self)
 		{
-			if (IsTraitDisabled)
+			if (IsTraitDisabled || actions.Count == 0)
 				return;
 
+			IEnumerable<int> modifiers = null;
 			foreach (var a in actions)
 			{
 				if (a.Delay-- <= 0)
 				{
-					var modifiers = self.TraitsImplementing<IDamageModifier>()
-						.Concat(self.Owner.PlayerActor.TraitsImplementing<IDamageModifier>())
-						.Select(t => t.GetDamageModifier(self, null));
+					if (modifiers == null)
+						modifiers = self.TraitsImplementing<IDamageModifier>()
+							.Concat(self.Owner.PlayerActor.TraitsImplementing<IDamageModifier>())
+							.Select(t => t.GetDamageModifier(self, null));
```

Comment:
You should call `ToArray` on this to cache the results. Otherwise you're caching the query outside the loop, but re-evaluating that query each time within the loop.

Reply:
Why are we regenerating this each tick anyway? What is wrong with my suggestion in https://github.com/OpenRA/OpenRA/pull/20131#discussion_r923905344?

Reply:
Something went wrong when pulling the changes you proposed from a repo on my other hosts. So i pushed the old version.

Hereby the correct changes.